commons-profiler-api/src/main/java/ru/fix/commons/profiler/impl/ProfilerReporterImpl.java

```diff
@@ -80,13 +81,21 @@ public void applyToSharedCounters(String profiledCallName, Consumer<SharedCounte
 
     @Override
     public ProfilerReport buildReportAndReset() {
+        return buildReportAndReset(Optional.empty());
+    }
+
+    @Override
+    public ProfilerReport buildReportAndReset(Optional<List<Pattern>> patterns) {
```

Comment:
maybe it would be better to use:  
public ProfilerReport buildReportAndReset() 
public ProfilerReport buildReportAndReset(List<Pattern> patterns) 
private ProfilerReport buildReportAndReset(Optional<List<Pattern>> patterns) 

and we probably should specify behavior in javadoc for case  buildReportAndReset(emptyList())

Reply:
ok